src/utils/auth.js

```diff
@@ -12,9 +13,14 @@ function newToken(id) {
 
 async function signup(req, res) {
   try {
+    //  save user
     const userSaved = await User.create(req.body)
+    //  create JWT Token
     const token = newToken(userSaved._id)
-    return res.status(200).json({ token })
+    //  create trainer
+    const trainerSave = await Trainer.create(req.body.trainer)
```

Comment:
By doing this, a trainer is created everytime /signup/ route is hitted but it's not working for the case when we want to simply create a client account (that is to say not creating a trainer account).
```
if(singupForTraineraccount) {
  const trainerSave = await Trainer.create(req.body.trainer)
}
```
Also, The following writes directly to MongoDB and should usually be avoided unless certain cases like writing tests or other cases when you deliberately want to write in invalid data.
`Trainer.create(req.body.trainer)`

Reply:
Yes, I agree we don't want to create a trainer every time a user signs up.  I've put that there just for testing purposes.  

I'll update the code to reflect your comments.  😊